scalameta/trees/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala

```diff
@@ -364,6 +364,9 @@ object TreeSyntax {
         case t: Pat.Xml              =>
           val zipped = t.parts.zip(t.args).map{ case (part, arg) => s(part, "${", arg, "}") }
           m(SimplePattern, s(r(zipped), t.parts.last))
+        case Pat.Typed(lhs, rhs : Lit) =>
```

Comment:
I will try this afternoon/night how to improve the `Literal`/`RefineTyp` selection mechanism.